sky/skylet/providers/command_runner.py

```diff
@@ -133,16 +135,33 @@ def run_init(self, *, as_head: bool, file_mounts: Dict[str, str],
             if not specific_image.startswith(server_prefix):
                 specific_image = f'{server_prefix}{specific_image}'
 
-        if self.docker_config.get('pull_before_run', True):
-            assert specific_image, ('Image must be included in config if ' +
-                                    'pull_before_run is specified')
-            self.run('{} pull {}'.format(self.docker_cmd, specific_image),
-                     run_env='host')
-        else:
-
-            self.run(f'{self.docker_cmd} image inspect {specific_image} '
-                     '1> /dev/null  2>&1 || '
-                     f'{self.docker_cmd} pull {specific_image}')
+        cnt = 0
+        max_retry = 5
+        while True:
+            try:
+                if self.docker_config.get('pull_before_run', True):
+                    assert specific_image, (
+                        'Image must be included in config if '
+                        'pull_before_run is specified')
+                    self.run('{} pull {}'.format(self.docker_cmd,
+                                                 specific_image),
+                             run_env='host')
+                else:
+                    self.run(
+                        f'{self.docker_cmd} image inspect {specific_image} '
+                        '1> /dev/null  2>&1 || '
+                        f'{self.docker_cmd} pull {specific_image}')
+                break
+            except click.ClickException as e:
```

Comment:
why is it click.ClickException?

Reply:
It is the exception raised by the underlying command runner when the command fails. It might be a better practice to only catch the exception that is raised by the underlying function.